nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1ClientImpl.java

```diff
@@ -16,59 +16,65 @@
-    private final int connectionQueueSize;
+    private static final Tls EMPTY_TLS = Tls.builder().build();
+    private final Http1ClientConfig clientConfig;
+
+    Http1ClientImpl(Http1ClientConfig clientConfig) {
+        super(Http1Client.builder()
+                      .update(it -> clientConfig.baseUri().ifPresent(it::baseUri))
+                      .update(it -> clientConfig.tls().ifPresent(it::tls))
+                      .channelOptions(clientConfig.socketOptions())
+                      .dnsResolver(clientConfig.dnsResolver())
+                      .dnsAddressLookup(clientConfig.dnsAddressLookup()));
 
-    Http1ClientImpl(Http1ClientBuilder builder) {
-        super(builder);
-        this.maxHeaderSize = builder.maxHeaderSize();
-        this.maxStatusLineLength = builder.maxStatusLineLength();
-        this.sendExpect100Continue = builder.sendExpect100Continue();
-        this.validateHeaders = builder.validateHeaders();
-        this.mediaContext = builder.mediaContext();
-        this.connectionQueueSize = builder.connectionQueueSize();
+        this.clientConfig = Http1ClientConfigDefault.toBuilder(clientConfig)
+                .services(Set.of()) // reset services to empty list
+                .update(it -> services(clientConfig).forEach(it::addService)) // add all configured services
+                .update(it -> it.tls(it.tls().orElse(EMPTY_TLS)))
+                .build();
     }
 
     @Override
     public Http1ClientRequest method(Http.Method method) {
         UriQueryWriteable query = UriQueryWriteable.create();
         UriHelper helper = (uri() == null) ? UriHelper.create() : UriHelper.create(uri(), query);
 
-        return new ClientRequestImpl(this, method, helper, query);
-    }
-
-    int maxHeaderSize() {
-        return this.maxHeaderSize;
+        return new ClientRequestImpl(clientConfig, method, helper, query);
     }
 
-    int maxStatusLineLength() {
-        return this.maxStatusLineLength;
+    Http1ClientConfig clientConfig() {
+        return clientConfig;
     }
 
-    boolean sendExpect100Continue() {
-        return this.sendExpect100Continue;
-    }
-
-    boolean validateHeaders() {
-        return this.validateHeaders;
-    }
+    private static List<WebClientService> services(Http1ClientConfig clientConfig) {
+        Config empty = Config.empty();
+        List<WebClientService> services = new ArrayList<>(clientConfig.services());
 
-    MediaContext mediaContext() {
-        return this.mediaContext;
-    }
+        services.addAll(HelidonServiceLoader
+                                .builder(ServiceLoader.load(WebClientServiceProvider.class))
+                                .useSystemServiceLoader(clientConfig.servicesUseServiceLoader())
+                                .build()
+                                .asList()
+                                .stream()
+                                .map(it -> it.create(empty))
+                                .toList());
```

Comment:
We should add `.stream()` to `HelidonServiceLoader` this pattern pops up quite often

Reply:
Yes